File: specs/esm/us-county-map.js
import * as vg from "@uwdata/vgplot";

await vg.coordinator().exec([
  vg.loadExtension("spatial"),
  vg.loadSpatial("counties", "data/us-counties-10m.json", {layer: "counties"}),
  vg.loadSpatial("states", "data/us-counties-10m.json", {layer: "states"})
]);

export default vg.plot(
  vg.geo(
    vg.from("counties"),
    {stroke: "currentColor", strokeWidth: 0.25}
  ),
  vg.geo(
    vg.from("states"),
    {stroke: "currentColor", strokeWidth: 1}
  ),
  vg.dot(
    vg.from("counties"),
    {
      x: vg.centroidX("geom"),
      y: vg.centroidY("geom"),
      r: 2,
      fill: "transparent",
      tip: true,
      title: "name"
    }
  ),
  vg.margin(0),
  vg.projectionType("albers")
);

File: docs/public/specs/python/us-county-map.py
import vgplot as vg

counties = vg.spatial("data/us-counties-10m.json", layer="counties")
states = vg.spatial("data/us-counties-10m.json", layer="states")

view = vg.plot(
    vg.geo(counties, stroke="currentColor", stroke_width=0.25),
    vg.geo(states, stroke="currentColor", stroke_width=1),
    vg.dot(
        counties,
        x=vg.centroid_x("geom"),
        y=vg.centroid_y("geom"),
        r=2,
        fill="transparent",
        tip=True,
        title="name",
    ),
    vg.margin(0),
    vg.projection_type("albers"),
)
